embedchain/embedchain.py

```diff
@@ -81,53 +82,104 @@ def add_local(self, data_type, content, metadata=None, config: AddConfig = None)
-            ids=ids,
-            where=where,  # optional filter
-        )
-
-        if len(existing_ids):
-            data_dict = {id: (doc, meta) for id, doc, meta in zip(ids, documents, metadatas)}
-            data_dict = {id: value for id, value in data_dict.items() if id not in existing_ids}
-
-            if not data_dict:
-                print(f"All data from {src} already exists in the database.")
-                return
-
-            ids = list(data_dict.keys())
-            documents, metadatas = zip(*data_dict.values())
-
-        # Add app id in metadatas so that they can be queried on later
-        if self.config.id is not None:
-            metadatas = [{**m, "app_id": self.config.id} for m in metadatas]
-
-        # FIXME: Fix the error handling logic when metadatas or metadata is None
-        metadatas = metadatas if metadatas else []
-        metadata = metadata if metadata else {}
-        chunks_before_addition = self.count()
-
-        # Add metadata to each document
-        metadatas_with_metadata = [{**meta, **metadata} for meta in metadatas]
+            try:
+                # get existing ids, and discard doc if any common id exist.
+                where = {"app_id": self.config.id} if self.config.id is not None else {}
+                # where={"url": src}
+                existing_docs = self.collection.get(
+                    ids=ids,
+                    where=where,  # optional filter
+                )
+                existing_ids = set(existing_docs["ids"])
+
+                if len(existing_ids):
+                    data_dict = {id: (doc, meta) for id, doc, meta in zip(ids, documents, metadatas)}
+                    data_dict = {id: value for id, value in data_dict.items() if id not in existing_ids}
+
+                    if not data_dict:
+                        logging.warning(f"All data from {src} already exists in the database.")
+                        return
+
+                    ids = list(data_dict.keys())
+                    documents, metadatas = zip(*data_dict.values())
+
+                # Add app id in metadatas so that they can be queried on later
+                if self.config.id is not None:
+                    metadatas = [{**m, "app_id": self.config.id} for m in metadatas]
+
+                # FIXME: Fix the error handling logic when metadatas or metadata is None
+                metadatas = metadatas if metadatas else []
+                metadata = metadata if metadata else {}
+                chunks_before_addition = self.count()
+
+                logging.log(100,"about to add metadata!")
+                # Add metadata to each document
+                metadatas_with_metadata = [{**meta, **metadata} for meta in metadatas]
+
+                self.collection.add(documents=documents, metadatas=list(metadatas_with_metadata), ids=ids)
+                logging.log(100,(f"Successfully saved {src}. New chunks count: " f"{self.count() - chunks_before_addition}"))
```

Comment:
why not use a regular loglevel like `debug` or `info`? https://docs.python.org/3/library/logging.html#logging-levels